Fix frontend build & test action

the frontend build & test action is currently broken

failed run: https://github.com/grovecj/mlb-stats/actions/runs/21336271349/job/61408634844?pr=7

```
Run npm test -- --run --passWithNoTests
  npm test -- --run --passWithNoTests
  shell: /usr/bin/bash -e {0}
npm error Missing script: "test"
npm error
npm error To see a list of scripts, run:
npm error   npm run
npm error A complete log of this run can be found in: /home/runner/.npm/_logs/2026-01-25T17_03_13_731Z-debug-0.log
Error: Process completed with exit code 1.
```

--- a/frontend/package.json
+++ b/frontend/package.json
@@ -7,7 +7,8 @@
     "dev": "vite",
     "build": "tsc && vite build",
     "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
-    "preview": "vite preview"
+    "preview": "vite preview",
+    "test": "vitest"
   },
   "dependencies": {
     "react": "^18.2.0",
@@ -24,6 +25,7 @@
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-react-refresh": "^0.4.5",
     "typescript": "^5.2.2",
-    "vite": "^5.0.8"
+    "vite": "^5.0.8",
+    "vitest": "^4.0.18"
   }
 }

